### PR TITLE
Merge SampleWebsiteTheme with Website app

### PR DESCRIPTION
**Pull request by PatrykSzwer**
For https://github.com/Starcounter/RebelsLounge/issues/223
First step for roadmap: https://github.com/Starcounter/RebelsLounge/issues/232

### CODE REVIEW

File: src/WebsiteEditor/Helpers/DataHelper.cs [hunk 1]
```diff
@@ -200,17 +206,38 @@ private static void GenerateBlendingPointPreview(WebSection section, WebUrl prev
 
         private static void GenerateTopbarSurface()
         {
-            var surface = new WebTemplate
+            var surface = new WebTemplate // Creates Login surface
```
Comment (un-tone) [inline src/WebsiteEditor/Helpers/DataHelper.cs]: // Creates Topbar surface

File: src/WebsiteEditor/Helpers/DataHelper.cs [hunk 2]
```diff
@@ -200,17 +206,38 @@ private static void GenerateBlendingPointPreview(WebSection section, WebUrl prev
...
+            {
+                Template = surface,
+                Name = "TopbarLeft",
+                Default = false
+            };
+
+            var topbarLeftCorner = new WebSection
+            {
+                Template = surface,
+                Name = "TopBarLeftCorner",
```
Comment (un-tone) [inline src/WebsiteEditor/Helpers/DataHelper.cs]: The other points have name started with "Topbar" but this one with "TopBar". Let's use one style.

File: src/WebsiteEditor/Helpers/DataHelper.cs [hunk 3]
```diff
@@ -221,21 +248,149 @@ private static void GenerateTopbarSurface()
...
+            new WebUrl // "Route all" route
             {
                 Template = surface,
                 Url = null,
                 IsFinal = true,
                 SortNumber = 100
             };
 
-            new WebMap { Section = topbar, ForeignUrl = "/signin/user", SortNumber = 1 };
+            var content = Db.SQL<Content>($"SELECT c FROM {typeof(Content).FullName} c WHERE {nameof(Content.URL)} = ?", DefaultSystemLogoContentUrl).FirstOrDefault();
```
Comment (un-tone) [inline src/WebsiteEditor/Helpers/DataHelper.cs]: Should be `c.{nameof(Content.URL)}` (the alias is missed)

Comment (un-tone) [reply src/WebsiteEditor/Helpers/DataHelper.cs]: You can write just `{typeof(Content)}` without `FullName` property. It looks more readable.

Comment (un-tone) [reply src/WebsiteEditor/Helpers/DataHelper.cs]: The same for the query below https://github.com/Starcounter/Website/pull/296/files#diff-791d27d90e41e68a4dbfe0c5512476a6R270

File: src/WebsiteEditor/Helpers/DataHelper.cs [hunk 4]
```diff
@@ -221,21 +248,149 @@ private static void GenerateTopbarSurface()
...
+            if (content == null)
```
Comment (un-tone) [inline src/WebsiteEditor/Helpers/DataHelper.cs]: I guess here you can use `??` operator like below

File: src/WebsiteEditor/Content/HtmlCompositions/login-composition.html [hunk 5]
```diff
@@ -0,0 +1,60 @@
+﻿<style>
+    .login-theme {
+        display: flex;
+        flex-flow: column;
+        align-items: center;
+    }
+</style>
+<script>
+    var script = document._currentScript || document.currentScript;
```
Comment (tomalec) [inline src/WebsiteEditor/Content/HtmlCompositions/login-composition.html]: It is unused.

File: src/WebsiteEditor/Content/HtmlCompositions/login-composition.html [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    var css =
```
Comment (tomalec) [inline src/WebsiteEditor/Content/HtmlCompositions/login-composition.html]: This pollutes global scope

File: src/WebsiteEditor/Content/HtmlCompositions/login-composition.html [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        }
+
+        .btn-link:hover, 
+        .btn-link:active:hover {
+            background-color: transparent;
+            color: #778899;
+        }`;
+    var style = document.createElement('style');
+    style.innerHTML = css;
+    document.head.appendChild(style);
```
Comment (tomalec) [inline src/WebsiteEditor/Content/HtmlCompositions/login-composition.html]: Are you stamping it to the global light dom consciously? Why don't you remove it when shadow DOM is removed?

File: src/WebsiteEditor/Content/HtmlCompositions/topbar-composition.html [hunk 8]
```diff
@@ -1,60 +1,186 @@
...
-     .website-topbarsurface-top {
-         background-color: #333;
-         color: #FFF;
-         padding: 10px;
-     }
-
-    polyfill-next-selector { content: '.website-topbarsurface-topbar > starcounter-include'; }/* for polyfilled browsers*/
-    .website-topbarsurface-topbar,
-    .website-topbarsurface-topbar > ::slotted(starcounter-include) {
+﻿<style>:host{display:block}</style>
```
Comment (tomalec) [inline src/WebsiteEditor/Content/HtmlCompositions/topbar-composition.html]: You can merge those two `<style>`s

File: src/WebsiteEditor/Content/HtmlCompositions/topbar-composition.html [hunk 9]
```diff
@@ -1,60 +1,186 @@
...
+    }
+
+    polyfill-next-selector { content: '.topbar-theme__top-center > starcounter-include'; }/* for polyfilled browsers*/
+    .topbar-theme__top-center ::slotted(starcounter-include) {
         display: flex;
         align-items: center;
         justify-content: flex-end;
     }
-    /* For ones with CSS Grid us it with 5px gutter */
-    @supports( (display: grid) and (grid-column-gap: 5px) ) {
```
Comment (tomalec) [inline src/WebsiteEditor/Content/HtmlCompositions/topbar-composition.html]: Are we redesigning it to something different so we drop the grid layout?

File: src/WebsiteEditor/Content/HtmlCompositions/topbar-composition.html [hunk 10]
```diff
@@ -1,60 +1,186 @@
...
-        }`;
-    var style = document.createElement('style');
-    style.innerHTML = css;
-    document.head.appendChild(style);
-</script>
-<section class='website-topbarsurface-top'>
-    <div class='website-topbarsurface-topbar'>
-        <slot name='websiteeditor/Topbar'></slot>
+<dom-module id="topbar-menu">
+    <template>
```
Comment (tomalec) [inline src/WebsiteEditor/Content/HtmlCompositions/topbar-composition.html]: Have you tried it w/o template?

File: src/WebsiteEditor/Content/HtmlCompositions/topbar-composition.html [hunk 11]
```diff
@@ -1,60 +1,186 @@
...
+                    var menu = this.querySelector("[topbar-menu-panel]");
+                    var attr = menu.hasAttribute("is-visible");
+                    if (attr === false) {
+                        menu.setAttribute("is-visible", "");
+                    } else {
+                        menu.removeAttribute("is-visible");
+                    }
+                });
+            }
+        });
```
Comment (tomalec) [inline src/WebsiteEditor/Content/HtmlCompositions/topbar-composition.html]: Shouldn't clicking on the panel background close it? AFAIR It was working like this at some point.

File: src/WebsiteEditor/Content/HtmlCompositions/topbar-composition.html (hunk 9 above)
Comment (PatrykSzwer) [reply src/WebsiteEditor/Content/HtmlCompositions/topbar-composition.html]: It's from old topbar-composition. I've replaced it with new one from SampleWebsiteTheme and it's shown here as change but It was never there in a first place, in a surface from SampleWebsiteTheme.

**Pull request merged**